mesonbuild/mesonlib/universal.py

```diff
@@ -1893,6 +1893,11 @@ def get_wine_shortpath(winecmd: T.List[str], wine_paths: T.Sequence[str]) -> str
 
     wine_paths = list(OrderedSet(wine_paths))
 
+    # do we even need to bother shortening the path?
+    wine_path = ";".join(wine_paths)
+    if len(wine_path) < 2048:
```

Comment:
I think it can be 2048 exactly without problems? Down below we check for `> 2048` in order to error out if (after the bat script) it is still too long. So maybe this should be `<= 2048`?

Reply:
I was worried about the need of a trailing NUL, and I can't find the limit in the wine codebase. I did wonder if `winepath` might be a better way to shorten those paths anyway...

Reply:
Either way, shouldn't we be consistent? and allow `< 2048` in both places, or `<= 2048` in both places?

Reply:
I've added a follow up patch to limit the WINEPATH to 2048-1 chars.

Reply:
In my testing for devenv, the limit is actually 1024.

Reply:
Also devenv has a smarter code to shorten path, using relative to builddir paths.